pandas/tools/pivot.py

```diff
@@ -81,9 +82,21 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
     DataFrame.pivot : pivot without aggregation that can handle
         non-numeric data
     """
+    pd_null = "_null_pd"
+
     index = _convert_by(index)
     columns = _convert_by(columns)
 
+    keys = index + columns
+
+    if not dropna:
+        key_data = np.array(data[keys], dtype='object')
```

Comment:
that is not acceptable, we use masks if needed. converting things like that will just lead to future reader confusion and bugs.